client-sdk/ts-web/core/src/hash.ts

```diff
@@ -1,5 +1,9 @@
-import {sha512_256} from 'js-sha512';
-
+/** Returns the SHA-512/256 hash of `data`. */
 export async function hash(data: Uint8Array) {
-    return new Uint8Array(sha512_256.arrayBuffer(data));
+    const dataBuf = data.buffer.slice(
+        data.byteOffset,
+        data.byteOffset + data.byteLength
+    );
+    const hashBuf = await crypto.subtle.digest("SHA-512", dataBuf);
```

Comment:
This is not right, it should be SHA-512/256 not SHA-512.

Reply:
SHA-512/256 is SHA-512 truncated to 256 bits, used because 64-bit 512 is more efficient on modern processors [[ref](https://eprint.iacr.org/2010/548.pdf))

Reply:
gets me every time

Reply:
ya, I wish we could use built in stuff. all this stuff is declared `async` in hope that one day we can